internal/git/gitsync_processor.go

```diff
@@ -601,7 +584,18 @@ func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeCli
 	}
 	localRepoPath := getLocalRepoPath(gitSync.GetName())
 	go func(repo *v1alpha1.RepositoryPath) {
-		r, err := cloneRepo(ctx, localRepoPath, repo)
+		globalConfig, err := controllerconfig.GetConfigManagerInstance().GetConfig()
+		if err != nil {
+			logger.Errorw("error getting  the  global config", "err", err)
+		}
+
+		gitCredentials := gitshared.FindCredByUrl(repo.RepoUrl, globalConfig)
+		method, err := gitshared.GetAuthMethod(ctx, gitCredentials, kubeClient, namespace)
+		if err != nil {
+			logger.Errorw("error getting  the  auth method", "err", err)
+		}
+
+		r, err := cloneRepo(ctx, localRepoPath, repo, method)
 		if err != nil {
```

Comment:
FYI, I have added [this](https://github.com/numaproj-labs/numaplane/issues/108) Issue to ensure we use the new Credentials in the case that Credentials change in the ConfigMap while we are reading from a repo (if we get an error). This is something that may make more sense to do after the Worker Queue refactor.

Reply:
So Do I need to change anything in here or should I leave it as it is for now @juliev0

Reply:
nothing to do now, thanks